warehouse/core/src/main/java/datawave/iterators/IteratorUtils.java

```diff
@@ -0,0 +1,139 @@
+package datawave.iterators;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
```

Comment:
Are all of these imports part of the "public API" ?

Reply:
AccumuloConfiguration is not public API in Accumulo.

Reply:
Taking a closer look, it seems *none* of these `org.apache.accumulo` imports are public API.

Reply:
I'd like to separate the issue of having Datawave work with Accumulo 2.0 from the issue of having Datawave only use classes in the Accumulo public API. I think the former is the more immediate goal and I think the latter will be unenforceable without a client only jar file. Going through all of the Datawave code now and making it work with *only* the Accumulo public API will be a ton of work that could easily be undone by someone coming in later and using a non-public API class or method because the accumulo-core jar is a maven dependency.